src/managers/input/InputManager.hpp

```diff
@@ -52,6 +53,7 @@ enum eBorderIconDirection : uint8_t {
 };
 
 struct STouchData {
+    SSessionLockSurface* touchFocusLockSurface;
```

Comment:
raw ptrs are banned in new code

Reply:
Now I remember why I didn't try to get this merged last year. :( Is a weak pointer suitable here?

Accordingly, Should this be changed to store shared pointers

https://github.com/hyprwm/Hyprland/blob/fda5626594c3f31bbab08e877a797a623c98c450/src/managers/SessionLockManager.hpp#L34

and this to return a shared pointer?

https://github.com/hyprwm/Hyprland/blob/fda5626594c3f31bbab08e877a797a623c98c450/src/managers/SessionLockManager.hpp#L52

Reply:
hm. I don't think they should be shared for no reason. I'll write a unique_ptr in hyprutils later today that allows weak pointers, if I don't by tomorrow remind me.
I've wanted one anyways.

Reply:
class is boring, here you go https://github.com/hyprwm/hyprutils/commit/423c69d697f56af4f8f2de7e2279eead17901228

Reply:
https://github.com/hyprwm/Hyprland/pull/9143

Reply:
What about this one here https://github.com/hyprwm/Hyprland/blob/ae403e6a05452336abd97e61be2706f6a930d827/src/managers/input/InputManager.cpp#L232

Should foundSurface be changed to a weak ptr?

Reply:
sls container should now store a UP if you rebase on main and you can make this func return a WP